core/src/message/context.rs

```diff
@@ -583,7 +582,8 @@ impl MessageContext {
     }
 
     /// Destructs context after execution and returns provided outcome and store.
-    pub fn drain(self) -> (ContextOutcome, ContextStore) {
+    pub fn drain(mut self) -> (ContextOutcome, ContextStore) {
+        self.store.local_nonce += self.tmp_store.outgoing.len() as u32;
```

Comment:
that's kind of dangerous, since we could adjust local nonce on each access, so I'd go this way